src/libraries/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs

```diff
@@ -107,23 +108,37 @@ public static void DangerousMakeTwosComplement(Span<uint> d)
             // where A = ~X and B = -Y
 
             // Trim trailing 0s (at the first in little endian array)
-            d = d.TrimStart(0u);
+            int i = d.IndexOfAnyExcept(0u);
 
-            // Make the first non-zero element to be two's complement
-            if (d.Length > 0)
+            if ((uint)i >= (uint)d.Length)
             {
-                d[0] = (uint)(-(int)d[0]);
-                d = d.Slice(1);
+                return;
             }
 
+            // Make the first non-zero element to be two's complement
+            d[i] = (uint)(-(int)d[i]);
+            d = d.Slice(i + 1);
+
             if (d.IsEmpty)
             {
                 return;
             }
 
-            // Make one's complement for other elements
-            int offset = 0;
+            DangerousMakeOnesComplement(d);
+        }
 
+        // Do an in-place one's complement. "Dangerous" because it causes
+        // a mutation and needs to be used with care for immutable types.
+        public static void DangerousMakeOnesComplement(Span<uint> d)
```

Comment:
If it accepts a non-readonly span - what is dangerous with it? We typically name such things with "Inplace"

Reply:
I followed the existing `DangerousMakeTwosComplement`. I have no strong preference for the naming convention, but if `DangerousMakeOnesComplement` is to be renamed, then `DangerousMakeTwosComplement` should be renamed as well.

https://github.com/dotnet/runtime/blob/8de6e0100dcd1b9e9a58ef3f3f877a8d5a0183db/src/libraries/System.Runtime.Numerics/src/System/Numerics/NumericsHelpers.cs#L100

Reply:
np. It just that it's the code that converts an immutable data into mutable span is what actually dangerous 🙂